test/test_archive_cleanup.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2022 Aiven, Helsinki, Finland. https://aiven.io/
+
+import datetime
+import json
+import sys
+from unittest import mock
+
+import pytest
+
+from pghoard import archive_cleanup
+
+
+def test_missing_config_arg():
+    args = ["archive_cleanup", "--dry-run"]
+    tool = archive_cleanup.ArchiveCleanup()
+    with mock.patch.object(sys, "argv", args):
+        assert tool.run() == 1
+
+
+def test_main_invalid_config():
+    args = ["archive_cleanup", "--config", "/etc/os-release", "--dry-run"]
+    with mock.patch.object(sys, "argv", args):
+        assert archive_cleanup.main() == 1
+
+
+def test_invalid_config():
+    args = ["archive_cleanup", "--config", "/etc/os-release", "--dry-run"]
+    tool = archive_cleanup.ArchiveCleanup()
+    with mock.patch.object(sys, "argv", args):
+        with pytest.raises(archive_cleanup.InvalidConfigurationError):
+            tool.run()
+
+
+def test_set_config():
+    tool = archive_cleanup.ArchiveCleanup()
+    tool.set_config("pghoard.json", "default")
+    assert tool.site == "default"
+    assert "basebackup_count" in tool.backup_site
+    assert "backup_location" in tool.config
+    assert tool.storage is not None
+
+
+def test_archive_cleanup(tmp_path):
```

Comment:
The tool is run multiple time to check for different cases, this test should probably be split in 3-4 different tests (with a fixture for the shared parts).

Reply:
Thanks, will have a look.